cirrus/google_cloud/manager.py

```diff
@@ -375,21 +376,22 @@ def create_or_update_bucket(
         if not bucket_exists:
             bucket.create()
 
-        if public:
-            # update bucket iam policy with allAuthN users having read access
-            policy = bucket.get_iam_policy()
-            role = GooglePolicyRole('roles/storage.objectViewer')
-            policy[str(role)] = ['allAuthenticatedUsers']
-            bucket.set_iam_policy(policy)
-        else:
-            policy = bucket.get_iam_policy()
-            role = GooglePolicyRole('roles/storage.objectViewer')
-            if 'allAuthenticatedUsers' in policy.get(str(role)):
-                policy[str(role)].remove('allAuthenticatedUsers')
-            bucket.set_iam_policy(policy)
+        if public is not None:
+            if public:
+                # update bucket iam policy with allAuthN users having read access
+                policy = bucket.get_iam_policy()
```

Comment:
the `policy`, `role`, and `bucket` statements can go outside either branch (of the `if public:` / `else`):
```python
policy = bucket.get_iam_policy()
role = str(GooglePolicyRole('roles/storage.objectViewer'))
if public:
    policy[role] = ['allAuthenticatedUsers']
else:
    policy[role].remove('allAuthenticatedUsers')
bucket.set_iam_policy(policy)
```

Reply:
ah, good call! changing it